src/loaders/s3m_load.c

```diff
@@ -622,6 +620,16 @@ static int s3m_load(struct module_data *m, HIO_HANDLE * f, const int start)
 
 	m->quirk |= QUIRKS_ST3 | QUIRK_ARPMEM;
 	m->read_event_type = READ_EVENT_ST3;
+	m->mvolbase = 48;
+	m->mvol = sfh.mv & 0x7f;
```

Comment:
Note that in earlier ST3 versions, master volume used a range from 0 to 7 instead, with 0x10 being the stereo flag. Later ST3 versions try to convert this to the newer interpretation, even when they shouldn't... Feel free to take a peek at OpenMPT's S3M loader for the gory details.

Reply:
Pushed what *ought* to be a fix for this.